### DISCUSSION TITLE
A NET 8 API project that references a shared library does not start when calling "dotnet watch run".

### DISCUSSION THREAD

**Discussion by code64Zmafi0** [nanoframework/Home#1495] [Q&A]
I'm using a shared library to store data transfer contracts.

The library is referenced .NET 8 api service and nanoframeowork application.

API runs through Visual Studio, but does not run with "dotnet watch run" command, cannot find target (error MSB4057).

How can I solve this problem?

**Comment by josesimoes** [comment]
You can't run a .NET nanoFramework application with `dotnet` CLI. It was its own runtime and CLR. Only with a real MCU or the virtual nanoCLR.

**Reply by code64Zmafi0** [reply]
I don't launch .NET nanoFramework application from dotnet, i launch API project!

**Reply by Ellerbach** [reply]
Can you please give couple of lines of code/pseudo code and explain the flow and what is nano code, what is .NET 8 code? Or just create a small graphic with this and the flow so that you can help us understanding what is your need.

**Reply by code64Zmafi0** [reply]
Hello. 
![image](https://github.com/nanoframework/Home/assets/49468734/58a86cf4-6145-44cf-9c2b-06322b35dfef)
Here is a diagram of the solution projects.

SmartItem.Server is .NET 8 api service. SmartItem.Server project refers to SmartItem.Shared.Iot. SmartItem.Shared.Iot is class library of DTOs  creates by template "ClassLibrary (.NET nanoFramework)". 

![image](https://github.com/nanoframework/Home/assets/49468734/e5ded066-9908-4c30-a13a-12c66167e155)

Starting a SmartItem.Server service via command "dotnet watch run" causes an error, 
but launch via VisualStudio (F5) is successful.

![image](https://github.com/nanoframework/Home/assets/49468734/022fcebe-81ab-42ac-b181-7af0f0ac1b9a)

**Reply by Ellerbach** [reply]
you'll have to specify which project to run: "dotnet run SmartItem.Client.csproj"
The reason it's failing is that "dotnet" command is not working for nanoFramework. So just specify what you want to run and all will go fine. Other way: create a separate sln just for nanoFramework (you still can reference the shared project) and another for the full .NET solution.
